services/frontend-service/src/ui/Batch.test.tsx

```diff
@@ -193,6 +193,31 @@ describe('Confirmation Dialog Provider', () => {
         },
     ];
 
+    const sampleCartActions: BatchAction[] = [
+        sampleDeployAction,
+        {
+            action: {
+                $case: 'createEnvironmentLock',
+                createEnvironmentLock: {
+                    environment: 'dummy environment',
+                    lockId: '1234',
+                    message: 'hello',
+                },
+            },
+        },
+        {
+            action: {
+                $case: 'createEnvironmentApplicationLock',
+                createEnvironmentApplicationLock: {
+                    environment: 'dummy environment',
+                    application: 'dummy application',
+                    lockId: '1234',
+                    message: 'hello',
+                },
+            },
+        },
+    ];
+
```

Comment:
I have a question about the logic but it may be trivial, what happened if a user/developer does the following:
`deployApp : app1 -> createApplock : lock1 -> undeployApp: app1 -> some other actions`
Assuming that he will click ` add anyway ` in the third step, then he checkout, 
the third action will fail with its original error ( because we can't un-deploy locked app) right? are the remaining actions will be ignored?

Reply:
Yeah, I'm actually not sure. this logic is handled in the backend.  In line `225` the transformer fails and then propagates the error all the way back to the service I think. This will not push any actions though. so nothing will change in the repo and you'll just get a notification error in the UI.
https://github.com/freiheit-com/kuberpult/blob/14c145b81934018f3c95ee3904974699e64bcdae/services/cd-service/pkg/repository/repository.go#L218-L232